### DISCUSSION TITLE
Profiling Launch - Alpha/Beta | Your Feedback Needed

### DISCUSSION THREAD

**Discussion by therealarkin** [getsentry/sentry#42520] [Launch Feedback]
We recently launched Profiling by Sentry – currently available for free to try out. With Profiling, you can see the code (down to the exact file/line) that is causing a spike or latency issue. Using flame charts, you can spot hot code paths with the longest-running functions affecting your UI performance. Catching these functions also lets you optimize CPU usage to reduce battery drain and improve warm / cold starts.

Profiling is currently in alpha for Python and Node.js, and in beta for Android and iOS. Try it out by clicking directly into the new "Profiling" tab on your sidebar.

We'd love to hear your feedback! Drop a comment below with your thoughts, or email us at [profiling@sentry.io](mailto:profiling@sentry.io). Some questions to chew on:

1. What tools do you use to profile your code today (if at all)?
2. What application performance issues are most critical for you to identify and fix?
3. What does your workflow for performance monitoring look like today? (e.g. checking certain metrics or dashboards, setting up alerts, tracing across services)
4. How comfortable do you feel interpreting flamegraphs / flamecharts?
5. P.S. Check out our blogs on profiling for [Python](https://blog.sentry.io/2022/11/21/profiling-for-python/) and [Node.js](https://blog.sentry.io/2022/11/21/profiling-for-node-js/) to learn more.

P.P.S. You'll need to have Sentry Performance set up to use Profiling - but no fear, here's a refresher on how to [get started](https://docs.sentry.io/product/performance/getting-started/) using just a few lines of code.

**Comment by therealarkin** [comment]
Still super curious about 1-5!
One thing I noticed about #2 from talking recently with few customers that there are a bunch of SQL libraries like SQLAlchemy  that easily go unoptimized, and with Sentry profiling+performance, it is easy to detect and fix

**Comment by smeubank** [comment]
We are also looking into profiling for react-native!

https://github.com/getsentry/sentry-react-native/discussions/3009
